lib/Parser.js

```diff
@@ -383,7 +477,7 @@ Parser.prototype = {
         const format = results => {
             if (_.contains([this.TYPES.SIMPLE, this.TYPES.GET, this.TYPES.VALUE, this.TYPES.COLLECTION], ruleType)) {
                 if (Array.isArray(results) && rule.type !== 'array') {
-                    return results.join(rule.separator || ' ');
+                    return results.length === 1 ? results[0] : results.join(rule.separator || ' ');
```

Comment:
yeah, it's for booleans if you do not parse an array: `[false].join(' ') === 'false'`